pkg/backup/s3/s3.go

```diff
@@ -30,36 +28,54 @@ const (
 	v1 = "v1/"
 )
 
-// S3 represents AWS S3 service.
+// S3 is to translate operator's S3 storage logic (e.g. dir layout) into S3 native code.
 type S3 struct {
 	bucket string
 	prefix string
 	client *s3.S3
 }
 
+type Config struct {
+	Bucket      string
+	Namespace   string
+	ClusterName string
+	SessOpt     *session.Options
+	Client      *s3.S3
+}
+
+// New creates S3 client translator.
+// If Client is given, it will be used as s3 client.
```

Comment:
i actually dislike this logic.

can we do NewFromClient, NewFromSessionOpt and New() instead?